api/types/databaseserver.go

```diff
@@ -64,12 +64,16 @@ type DatabaseServer interface {
 	GetURI() string
 	// GetCA returns the database CA certificate bytes.
 	GetCA() []byte
-	// GetRegion returns the AWS region for RDS/Aurora databases.
-	GetRegion() string
+	// GetAWS returns AWS information for RDS/Aurora databases.
+	GetAWS() AWS
+	// GetGCP returns GCP information for Cloud SQL databases.
+	GetGCP() GCP
 	// GetType returns the database type, self-hosted or AWS RDS.
 	GetType() string
 	// IsRDS returns true if this is an RDS/Aurora database.
 	IsRDS() bool
```

Comment:
I think that `IsGCP` was actually a poor choice, renamed it to `IsCloudSQL`. When we add support for Redshift, it will technically also be "AWS" but use different auth type, so we'll likely have to add `IsRedshift`.